tests/unit/conditions/test_rpc_condition.py

```diff
@@ -133,3 +136,98 @@ def test_rpc_condition_invalid_comparator_value_type(invalid_value, rpc_conditio
                 value=invalid_value,
             ),
         )
+
+
+def test_rpc_condition_uses_provided_endpoint(mocker):
+    # Mock HTTPProvider
```

Comment:
```suggestion
    # Spy HTTPProvider
```

^ some other similar spots